### ISSUE TITLE
[rush] Respect pnpmfile.js when "rush deploy" calculates dependencies

### ISSUE THREAD

**Issue by arikon** [microsoft/rushstack#1941]
## Please prefix the issue title with the project name i.e. [rush], [api-extractor] etc. ##

**Is this a feature or a bug?**

- [x] Feature
- [ ] Bug

**Please describe the actual behavior.**

For now `rush deploy` does not use info from pnpmfile.js to copy dependencies.

For example:

pnpmfile.js
```js
"use strict";

module.exports = {
  hooks: {
    readPackage
  }
};

function readPackage(packageJson, context) {

  if (packageJson.name === '@nestjsx/crud') {
    packageJson.dependencies['@nestjs/common'] = '^7.1.3';
  }

  if (packageJson.name === '@nestjsx/crud-typeorm') {
    packageJson.dependencies['@nestjsx/util'] = '^4.6.2';
  }

  return packageJson;
}
```

`@nestjs/common` and `@nestjsx/util` will not be copied during `rush deploy`.

We could process every package.json that is read by rush deploy using `hooks.readPackage()` hook from `pnpmfile.js` to produce correct deploy bundle.

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**

**What is the expected behavior?**

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** 
* **Tool Version:** 
* **Node Version:** 
  * **Is this a LTS version?** 
  * **Have you tested on a LTS version?** 
* **OS:**

**Comment by octogonz** [comment]
Here's a PR: https://github.com/microsoft/rushstack/pull/1948

**Comment by octogonz** [comment]
This fix was released with Rush 5.27.0
